packages/apptive_grid_core/lib/network/apptive_grid_client.dart

```diff
@@ -82,16 +82,18 @@ class ApptiveGridClient {
   Future<FormData> loadForm({
     required FormUri formUri,
     Map<String, String> headers = const {},
+    bool isRetry = false,
   }) async {
-    var formHeaders = _createHeaderWithDefaults(headers);
     final url = _generateApptiveGridUri(formUri.uri);
-    final response = await _client.get(url, headers: formHeaders);
+    final response =
+        await _client.get(url, headers: _createHeadersWithDefaults(headers));
     if (response.statusCode >= 400) {
-      if (response.statusCode == 401) {
+      if (response.statusCode == 401 && isRetry) {
         await _authenticator.checkAuthentication();
         return loadForm(
           formUri: formUri,
-          headers: formHeaders,
+          headers: headers,
```

Comment:
```suggestion
          headers: _createHeadersWithDefaults(headers),
```

Reply:
this will be called in the recursion anyways